pulsar-client/src/main/java/org/apache/pulsar/client/impl/Backoff.java

```diff
@@ -84,19 +121,30 @@ public long next() {
         return Math.max(initial, current);
     }
 
-    public void reduceToHalf() {
+    /**
+     * Reduce the next backoff to half of its current value. If the result is less than {@link #initial}, set
+     * {@link #next} to {@link #initial}.
+     */
+    public synchronized void reduceToHalf() {
         if (next > initial) {
             this.next = Math.max(this.next / 2, this.initial);
         }
     }
 
-    public void reset() {
+    /**
+     * Reset this {@link Backoff} instance to its initial state.
+     */
+    public synchronized void reset() {
         this.next = this.initial;
         this.mandatoryStopMade = false;
     }
 
+    public synchronized boolean isMandatoryStopMade() {
```

Comment:
This method looks like just getting the current value, should we synchronize it?

Reply:
We either need to synchronize it or make the variable volatile. Since it is only used in one place (and is possibly used incorrectly), I lean towards making it synchronized, for now.